### DISCUSSION TITLE
How is this compared to Modula 2?

### DISCUSSION THREAD

**Discussion by ghost** [modula3/cm3#1056] [General]
What is the different between Modula 3 language and Modula 2? Can I do sth useful with this? With Modula 2, I have the ADW compiler to develop native Win32 app. If I want more performance and develop cli app only then GNU Modula 2 is rock solid. How is this compared to Modula 2 at all?

**Comment by mikanystrom** [comment]
Modula-3 is a cleaned up Modula-2+.

There's a report from DEC about what Modula-2+ is and why it was invented.

https://hopl.info/showlanguage.prx?exp=1101&language=Modula-2+

Look for SRC-RR-3 ... Here's one source

https://archive.org/details/bitsavers_dectechrep_1876256
